### PR TITLE
Add oVirt ovirt_templates and ovirt_templates_facts modules

### PR DESCRIPTION
**Pull request by machacekondra**
##### ISSUE TYPE

- New Module Pull Request
##### COMPONENT NAME

- ovirt_templates
- ovirt_templates_facts
##### SUMMARY

This PR adds new modules for managing oVirt templates.

### CODE REVIEW

**Comment by machacekondra** [comment]
@mwperina @ryansb

**Comment by gregdek** [comment]
Thanks again to @machacekondra. This module is going into community review.

We notice that you are one of the maintainers (@machacekondra) of this module, so when you think it's ready, please comment "shipit" and we will consider it for merging.

[This message brought to you by your friendly Ansibull-bot.]

File: cloud/ovirt/ovirt_templates.py [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+
+
+DOCUMENTATION = '''
+---
+module: ovirt_templates
+short_description: Module to manage virtual machine templates in oVirt
+version_added: "2.3"
+author: "Ondra Machacek (@machacekondra)"
+description:
+    - "Module to manage virtual machine templates in oVir."
```
Comment (mwperina) [inline cloud/ovirt/ovirt_templates.py]: oVir -> oVirt

Comment (mwperina) [reply cloud/ovirt/ovirt_templates.py]: virtual machine -> VM

File: cloud/ovirt/ovirt_templates_facts.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+notes:
+    - "This module creates a new top-level C(ovirt_templates) fact, which
+       contains a list of templates."
+options:
+    pattern:
+      description:
+        - "Search term which is accepted by oVirt search backend."
+        - "For example to search template X from datacenter Y use following pattern:
+           name=X and datacenter=Y"
+extends_documentation_fragment: ovirt
```
Comment (mwperina) [inline cloud/ovirt/ovirt_templates_facts.py:50]: ovirt -> oVirt

File: cloud/ovirt/ovirt_templates_facts.py [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+'''
+
+EXAMPLES = '''
+# Examples don't contain auth parameter for simplicity,
+# look at ovirt_auth module to see how to reuse authentication:
+
+# Gather facts about all templates named C<rhel**> from datacenter C(west):
```
Comment (mwperina) [inline cloud/ovirt/ovirt_templates_facts.py]: I'd rephrase to: 'Gather facts about all templates which names start with C<centos> and belongs to data center C(west):'

File: cloud/ovirt/ovirt_templates_facts.py [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+- ovirt_templates_facts:
+    pattern: name=rhel* and datacenter=west
```
Comment (mwperina) [inline cloud/ovirt/ovirt_templates_facts.py]: rhel -> centos

**Review by mwperina — CHANGES_REQUESTED** [review]

File: cloud/ovirt/ovirt_templates_facts.py [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+        - "For example to search template X from datacenter Y use following pattern:
+           name=X and datacenter=Y"
+extends_documentation_fragment: ovirt
+'''
+
+EXAMPLES = '''
+# Examples don't contain auth parameter for simplicity,
+# look at ovirt_auth module to see how to reuse authentication:
+
+# Gather facts about all templates which names start with C(centos*) and belongs to data center C(west):
```
Comment (mwperina) [inline cloud/ovirt/ovirt_templates_facts.py]: 'C(centos*)' -> 'C(centos)'

**Review by mwperina — CHANGES_REQUESTED** [review]

**Review by mwperina — APPROVED** [review]
Looks good to me, thanks! :+1:

**Comment by machacekondra** [comment]
shipit

**Comment by gregdek** [comment]
Thanks @machacekondra. Since you are one of the maintainers (@machacekondra) of this module and you commented with "shipit", we are marking this PR for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
Thanks again to @machacekondra. This module is going into community review.

We notice that you are one of the maintainers (@machacekondra) of this module, so when you think it's ready, please comment "shipit" and we will consider it for merging.

[This message brought to you by your friendly Ansibull-bot.]

File: cloud/ovirt/ovirt_templates.py [hunk 6]
```diff
@@ -0,0 +1,299 @@
...
+                template = wait_for_import(module, templates_service)
+                ret = {
+                    'changed': True,
+                    'id': template.id,
+                    'template': get_dict_of_struct(template),
+                }
+
+        module.exit_json(**ret)
+    except Exception as e:
+        module.fail_json(msg=str(e))
```
Comment (ryansb) [inline cloud/ovirt/ovirt_templates.py]: Same here, please surface the full exception.

File: cloud/ovirt/ovirt_templates_facts.py [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+        module.exit_json(
+            changed=False,
+            ansible_facts=dict(
+                ovirt_templates=[
+                    get_dict_of_struct(c) for c in templates
+                ],
+            ),
+        )
+    except Exception as e:
+        module.fail_json(msg=str(e))
```
Comment (ryansb) [inline cloud/ovirt/ovirt_templates_facts.py]: Same here, please surface the full exception.

**Comment by machacekondra** [comment]
shipit

**Comment by gregdek** [comment]
Thanks @machacekondra. Since you are one of the maintainers (@machacekondra) of this module and you commented with "shipit", we are marking this PR for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

**Review by mwperina — APPROVED** [review]
:+1:

File: cloud/ovirt/ovirt_templates_facts.py (hunk 2 above)
Comment (ryansb) [reply cloud/ovirt/ovirt_templates_facts.py:50]: @mwperina Lower case is the correct reference.
